lighthouse-core/audits/seo/crawlable-anchors.js

```diff
@@ -0,0 +1,86 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether anchors have hyperlinks which can be crawled by search engines. This descriptive title is shown when all hyperlinks on the page are crawlable. */
+  title: 'Anchors have crawlable hyperlinks',
+  /** Descriptive title of a Lighthouse audit that provides detail on whether anchors have hyperlinks which can be crawled by search engines. This descriptive title is shown when there are hyperlinks which are not crawlable by search engines. */
+  failureTitle: 'Anchors do not have crawlable hyperlinks',
+  /** Description of a Lighthouse audit that tells the user why hyperlinks should be crawlable. This is displayed after a user expands the section to see more. 'Learn More' becomes link text to additional documentation. */
+  description: 'Search engines use hyperlinks to crawl websites',
+  /** Label for a column in a data table; entries will be the HTML anchor elements that failed the audit. Anchors are DOM elements that are links. */
+  columnFailingAnchors: 'Failing Anchor Elements',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class CrawlableAnchors extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'crawlable-anchors',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['AnchorElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit({AnchorElements: anchorElements}) {
+    const failingAnchors = anchorElements.filter(({rawHref, listeners = [], name = ''}) => {
+      if (rawHref.startsWith('file:')) {
+        return true;
+      }
+
+      const hasClickHandler = listeners.some(({type}) => type === 'click');
+
+      if (hasClickHandler || name.trim().length > 0) return;
+
+      if (rawHref === '') {
+        return true;
+      }
+
+      if (rawHref.startsWith('javascript:void(0)')) {
+        return true;
+      }
+    });
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [{
+      key: 'node',
+      itemType: 'node',
+      text: str_(UIStrings.columnFailingAnchors),
+    }];
+
+    /** @type {LH.Audit.Details.Table['items']} */
+    const itemsToDisplay = failingAnchors.map(node => {
+      return {
+        node: {
+          type: 'node',
```

Comment:
we need to do a little more to make this a clickable node in devtools, see 

https://github.com/GoogleChrome/lighthouse/blob/82f20d777e320083d846db1794e1476a15c20310/lighthouse-core/audits/dobetterweb/external-anchors-use-rel-noopener.js#L72-L78

for how they construct their nodes.

Reply:
Oh cool that's help! Have use that 👍️ thanks for linking to that example